src/components/BaseLayerSwitcher/BaseLayerSwitcher.js

```diff
@@ -0,0 +1,178 @@
+/* eslint-disable jsx-a11y/interactive-supports-focus */
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { FaChevronCircleLeft } from 'react-icons/fa';
+import Layer from '../../layers/Layer';
+import img from '../../images/baselayer/osm.baselayer.png';
+
+import './BaseLayerSwitcher.scss';
+
+const propTypes = {
+  /**
+   * An array of react-spatial layers.
+   */
+  layers: PropTypes.arrayOf(PropTypes.instanceOf(Layer)).isRequired,
+
+  /**
+   * Object containing relative paths to the base layer images.
+   */
+  layerImages: PropTypes.objectOf(PropTypes.string),
+
+  /**
+   * CSS class to apply on the container.
+   */
+  className: PropTypes.string,
+
+  /**
+   * Alternaive text rendered if layer images can't be loaded
+   */
+  altText: PropTypes.string,
+
+  /**
+   * Button titles.
+   */
+  titles: PropTypes.shape({
+    button: PropTypes.string,
+    openSwitcher: PropTypes.string,
+    closeSwitcher: PropTypes.string,
+  }),
+};
+
+const defaultProps = {
+  className: 'rs-base-layer-switcher',
+  altText: 'Source not found',
+  titles: {
+    button: 'Base layers',
+    openSwitcher: 'Open Baselayer-Switcher',
+    closeSwitcher: 'Close Baselayer-Switcher',
+  },
+  layerImages: undefined,
+};
+
+const getNextImage = (currentLayer, layers, layerImages) => {
+  const currentIndex = layers.indexOf(
+    layers.find(layer => layer === currentLayer),
+  );
+  const nextIndex = currentIndex + 1 === layers.length ? 0 : currentIndex + 1;
+  return layerImages.filter(layer => layer)[
+    Object.keys(layerImages)[nextIndex]
+  ];
```

Comment:
Is `layerImages[nextIndex]` not enough ?

Reply:
I love overcomplicating things XD Thanks